rumqttd/src/router/routing.rs

```diff
@@ -1358,7 +1370,23 @@ fn forward_device_data(
         inflight_slots = 1;
     }
 
-    let (next, publishes) =
+    let mut publishes = Vec::new();
+
+    if request.forward_retained_msg {
+        // NOTE: limit the number of read messages
+        // and skip the messages previously read.
```

Comment:
how do we skip previously read messages here?

Reply:
as of now, we don't / can't.